packages/api/src/lib/prepareParams.ts

```diff
@@ -105,165 +165,171 @@ export default async function prepareParams(operation: Operation, body?: unknown
       selected: number;
       variables: Record<string, string | number>;
     };
-  } = {};
+  } = jsonSchemaDefaults;
 
-  let shouldDigestParams = false;
-
-  if (Array.isArray(body)) {
-    // If the body param is an array, then it's absolutely a body and not something we need to do
-    // analysis against.
-    params.body = body;
-
-    if (typeof metadata !== 'undefined') {
-      shouldDigestParams = true;
-    }
-  } else if (typeof metadata === 'undefined') {
-    // No metadata was explicitly defined so we need to analyze the body to determine if it should
-    // actually be treated as metadata.
-    shouldDigestParams = true;
-  } else {
-    // Body and metadata were both supplied.
-    params.body = body;
-    shouldDigestParams = true;
-  }
-
-  let digested: Record<string, ParameterObject> = {};
-  let metadataIntersected = false;
-  let hasDigestedParams = false;
-  if (shouldDigestParams) {
-    digested = digestParameters(operation.getParameters());
-    hasDigestedParams = !!Object.keys(digested).length;
-  }
-
-  // No metadata was explicitly defined so we need to analyze the supplied, and we haven't already
-  // set a body then we need to analyze the supplied body to see if it should actually be metadata.
-  // If not, then we can just treat it as a body and pass it along.
-  if (!('body' in params) && typeof metadata === 'undefined') {
-    if (!hasDigestedParams) {
-      // No parameters were able to be digested, so we just have to assume that what the user
-      // supplied was for a body. This might lead to unwanted false positives if an OAS isn't
-      // accurate, but short of throwing an error there isn't anything we can really do about it.
-      params.body = body;
-    } else {
-      const intersection = Object.keys(body).filter(value => Object.keys(digested).includes(value)).length;
-      if (intersection && intersection / Object.keys(body).length > 0.25) {
-        // If more than 25% of the body intersects with the parameters that we've got on hand, then
-        // we should treat it as a metadata object and organize into parameters.
-        // eslint-disable-next-line no-param-reassign
-        metadata = body as Record<string, unknown>; // @todo we should just do a "if body isnt an object don't digest it"
-        metadataIntersected = true;
+  // If a body argument was supplied we need to do a bit of work to see if it's actually a body
+  // argument or metadata because the library lets you supply either a body, metadata, or body with
+  // metadata.
+  if (typeof body !== 'undefined') {
```

Comment:
I hope this code is a lot cleaner for you all to read and understand then what it was.